src/Abstractions/NexusMods.Abstractions.Serialization/Settings/Extensions.cs

```diff
@@ -0,0 +1,21 @@
+using NexusMods.Abstractions.Settings;
+
+namespace NexusMods.Abstractions.Serialization.Settings;
+
+/// <summary>
+/// Extension methods for settings.
+/// </summary>
+public static class Extensions
+{
+    internal static SettingsStorageBackendId StorageBackendId = SettingsStorageBackendId.From(Guid.Parse("6317967a-3ee9-4e3f-bb24-ebbe40560160"));
+
+    /// <summary>
+    /// Use the data store as a storage backend for this setting.
+    /// </summary>
+    public static ISettingsStorageBackendBuilder<T> UseDataStore<T>(
+        this ISettingsStorageBackendBuilder<T> builder)
+        where T : class, ISettings, new()
+    {
+        return builder.UseStorageBackend(StorageBackendId);
```

Comment:
I'm a bit confused with this. 
I don't think it makes sense to have the id defined in `Extensions.cs`.
Shouldn't we have `IDataStoreSettingsBackend` defined either here or some other Abstractions project, or an abstract class if it can't be an interface for some reason? That way the id is defined in there and is accessible.

Reply:
That'd only shift the internals around. The consumer would still call `UseDataStore<T>`.

Reply:
Wouldn't they use UseStorageBackend<IDataStoreSettingsBackend>() or something more along those lines?

Reply:
The `TelemetrySettings` use the extension method:

```csharp
.ConfigureStorageBackend<TelemetrySettings>(builder => builder.UseDataStore())
```

Reply:
Yeah, why do we need the extension method? I think it's a symptom that something doesn't quite make sense

Reply:
Maybe we can get in a call and you can give me an overview of how it's supposed to work